GetandTake/Pages/Account/Register.cshtml.cs

```diff
@@ -85,6 +89,18 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
             }
         }
 
+        if (!await _roleManager.RoleExistsAsync(Role.Admin))
+        {
+            await _roleManager.CreateAsync(new IdentityRole(Role.Admin));
+        }
+
+        if (!await _roleManager.RoleExistsAsync(Role.User))
+        {
+            await _roleManager.CreateAsync(new IdentityRole(Role.User));
+        }
```

Comment:
No need to create `Role` entities when you create user. First, it violates _Single Responsibility_ rule. Second, usually, if role does not exist, we should return error to user that something gone wrong. By default, there are some already predefined simplest roles like _User_, _Admin_ that can manage operations. Let's remove this part of code and assign to B2C (application end user) just _User_ role (admins are created via different app)